CYBRHardeningCheck/Main.ps1

```diff
@@ -187,13 +196,21 @@ Function EndScript
 	End the Script with a Footer line and Remove all used modules
 #>
 	param(
-		$moduleInfos
+		$moduleInfos,
+        [switch]$ReturnData,
+        [string]$HardeningReportPath
 	)
 
 	Write-LogMessage -Type Info -MSG "Script ended" -Footer -LogFile $LOG_FILE_PATH
 
 	# UnLoad loaded modules
-	Remove-ScriptModule $moduleInfos	
+	Remove-ScriptModule $moduleInfos
+
+    # Return the Report Path if ReturnData switch is selected	
+    if($ReturnData.IsPresent)
+    {
+        return $HardeningReportPath
```

Comment:
You return a value that you received as parameter, why?
If you already had it, use it

Reply:
@AssafMiron The parameter is $ReturnData (switch)
And the script will return the Report path instead of opening it.
It will allow to execute the script remotely through WinRm session and retrieve the report location to be able to copy it from PSSession. Example:
#region -------- EXAMPLE: remote execution script block to get the report path --------
$RunRemoteHardeningReport = {
    $ReportPath = S:\servertemp\scripts\Scripts\GLOBAL\CYBRHardeningCheck\Main.ps1 -Verbose -Debug -ReturnReportPath
    return $ReportPath
}

# execute remote scropt block and return the report file path
$RemoteReportPath = Invoke-Command -Session $mysession -ScriptBlock $RunRemoteHardeningReport

# Copy remote report file to the network path
Copy-Item -Path $RemoteReportPath -Destination $NetworkHardeningReportPath -Force -FromSession $mysession -Verbose

Reply:
I understand the use case and why
But specifically in this function you are not using it like this
The EndScript function is not responsible of opening the file, this happens in the main
So what happens is that you call in main to this EndScript function with a parameter of the HTML report and if the switch is returnData then you return the same HTML report path that I called this script with
but you don't need it, you just need to return that in the main script instead of opening the file (which you did)